src/Command/QueueCommand.php

```diff
@@ -73,23 +72,23 @@ protected function configure()
         try {
             // Transform into something for gearman.
             $entity = $this->transformer->transform($item, $this->serializedTransform);
-        } catch (BadResponse $e) {
-            // We got a 404 or server error.
-            $this->logger->error("{$this->getName()}: Item does not exist in API: {$item->getType()} ({$item->getId()})", [
-                'exception' => $e,
-                'item' => $item,
-            ]);
-            // Remove from queue.
-            $this->queue->commit($item);
         } catch (Throwable $e) {
-            // Unknown error.
-            $this->logger->error("{$this->getName()}: There was an unknown problem importing {$item->getType()} ({$item->getId()})", [
-                'exception' => $e,
-                'item' => $item,
-            ]);
-            $this->monitoring->recordException($e, "Error in importing {$item->getType()} {$item->getId()}");
-            // Remove from queue.
-            $this->queue->commit($item);
+            if ($e instanceof BadResponse && in_array($e->getResponse()->getStatusCode(), [404, 410])) {
+                $this->logger->error("{$this->getName()}: Item does not exist in API: {$item->getType()} ({$item->getId()})", [
+                    'exception' => $e,
+                    'item' => $item,
+                ]);
+                // Remove from queue.
+                $this->queue->commit($item);
```

Comment:
https://github.com/elifesciences/bus-sdk-php/pull/27#discussion_r152300994 still applies.

Reply:
Are services actually emitting events on deletion?

Reply:
They should do...